### ISSUE TITLE
Switch to SQLite

### ISSUE THREAD

**Issue by pheerai** [f3l/f3lcites#21]
The following Table-Layout is Advised (using UTF8 globally using `--default-character-set=utf8`)

Cites:

| int ID (PRIMARY, AUTO_INC) | text Cite | datetime ADDED | varchar(80) FROM |
| --- | --- | --- | --- |
| 1 | "." | 2016-04-26 12:00 | pheerai |

Changes:

| Int ID (PRIMARY, AUTO_INC) | int citeID | datetime CHANGED | varchar(80) BY |
| --- | --- | --- | --- |
| 1 | 1 | 2016-04-27 13:00 | pheerai |
| 2 | 1 | 2016-04-27 13:02 | pheerai |

Maybe Add views for IDs only and Cite + last Edit.

This is tightly coupled with F3LBot!

Edit: Switched to sqlite. The basic layout has changed, but most of it is still valid.

**Comment by pheerai** [comment]
Create cites using:

```
CREATE TABLE IF NOT EXISTS cites (
    id BIGINT UNSIGNED AUTO_INCREMENT,
    cite TEXT,
    added DATETIME,
    addedby VARCHAR(128),
    PRIMARY KEY (id)
);
```

Add new using:

```
INSERT INTO cites (cite, added, addedby) 
VALUES (
    ".", 
    '2016-04-11 13:42',
    'pheerai'
);
```

**Comment by pheerai** [comment]
Create changes using:

```
CREATE TABLE IF NOT EXISTS changes (
    id BIGINT UNSIGNED AUTO_INCREMENT,
    citeid BIGINT UNSIGNED,
    changed DATETIME,
    changedby VARCHAR(128),
    PRIMARY KEY(id)
);
```

Add new Change using:

```
INSERT INTO changes (citeid, changed, changedby)
VALUES (
    '2',
    '2016-04-11 13:47',
    'pheerai'
);
```

**Comment by pheerai** [comment]
Other usefull Commands:

Get entires 10 entries, starting from x:

```
SELECT * FROM cites ORDER BY added DESC LIMIT x, 10;
```

Get All cites with newest changed date:

```
CREATE VIEW mergecites AS SELECT cites.id, cites.cite, cites.added, cites.addedby, changes.changed, changes.changedby FROM cites LEFT JOIN changes ON cites.id = changes.citeid;
CREATE VIEW showcites AS SELECT id, cite, added, addedby, MAX(changed), changedby FROM mergecites GROUP BY id;
```

**Comment by pheerai** [comment]
Closed with 9be3323
